### PR TITLE
feat: add `EtlConfig` as well as setting the directory to datadir

### LINKED ISSUE #7054: sync crashing running out of space on /tmp
### Describe the bug

I'm trying to sync on the new db version (started with alpha20, and getting the same issue with beta1).

I have plenty of space, just not on /tmp:
```bash
df -h  
Filesystem      Size  Used Avail Use% Mounted on
/dev/nvme0n1p5  785G  229G  517G  31% /
devtmpfs        4.0M     0  4.0M   0% /dev
tmpfs            16G   42M   16G   1% /dev/shm
tmpfs           6.3G  2.1M  6.3G   1% /run
tmpfs            16G  128M   16G   1% /tmp
/dev/nvme2n1p1  3.6T  2.3T  1.2T  67% /nvme
/dev/sdf2       3.7T 1013G  2.7T  28% /seagate
/dev/nvme0n1p4  984G   36G  898G   4% /home
tmpfs           3.2G   34M  3.1G   2% /run/user/1000
```

I have no idea what /tmp is being used for. Just intermediate storage during sync, I'd image. If so, I'm guessing there's no startup flag I can use to avoid this. See my startup flags below.

### Steps to reproduce

this is my startup script:
```bash
# set logging level
logging_level="info,net=error,reth::node::events=error,consensus::engine=debug"

# execute the command
RUST_LOG=$logging_level $command node \
    --log.file.directory ./logs_mainnet \
    --datadir /nvme/reth_mainnet \
    --chain mainnet \
    --http \
    --http.api eth,debug,net,trace,web3,rpc
```

### Node logs

```text
2024-03-08T17:27:35.352146Z  INFO reth::cli: Status connected_peers=5 freelist=1447343 stage=TransactionLookup checkpoint=0 target=19384525 stage_progress=None
2024-03-08T17:27:45.576156Z  INFO reth_node_core::events::node: Forkchoice updated head_block_hash=0x51e127f36756be927cc0bfa510acb1b4f3040358fbc351cd49c6bf2cb71ea114 safe_block_hash=0x2519b014b64b9ed9e105a3aa6c6829b232212bf6e51166b4f4030b81a643cc16 finalize
d_block_hash=0xea91bc916876a0c5cc101ea4ea94a4cff5c39d0969a0ccd37f354aa65e151153 status=Syncing
2024-03-08T17:27:55.548176Z ERROR sync::pipeline: Stage encountered a fatal error: No space left on device (os error 28) at path "/tmp/.tmpwjHJZQ/.tmpqKmTT1" stage=TransactionLookup
2024-03-08T17:27:55.552249Z ERROR reth::cli: shutting down due to error
Error: No space left on device (os error 28) at path "/tmp/.tmpwjHJZQ/.tmpqKmTT1"
                                                                
Location:                  
    /rustc/9c3ad802d9b9633d60d3a74668eb1be819212d34/library/core/src/task/poll.rs:255:39
```

### Platform(s)

Linux (x86)

### What version/commit are you on?

reth Version: 0.2.0-beta.1
Commit SHA: 3c028e5ef
Build Timestamp: 2024-03-08T16:12:50.095325399Z
Build Features: jemalloc
Build Profile: maxperf

previous version I just upgraded from, which gives the same error:
reth Version: 0.1.0-alpha.21
Commit SHA: ef7ce3601
Build Timestamp: 2024-03-02T22:52:35.330153182Z
Build Features: jemalloc
Build Profile: maxperf

### What database version are you on?

Current database version: 2

### What type of node are you running?

Archive

### What prune config do you use, if any?

None, I don't have a reth.toml as far as I'm aware of.

### If you've built Reth from source, provide the full command you used

RUSTFLAGS="-C target-cpu=native" cargo build --profile maxperf --features jemalloc

### Code of Conduct

- [X] I agree to follow the Code of Conduct

### PR DESCRIPTION
**Pull request by joshieDo**
Built on top of https://github.com/paradigmxyz/reth/pull/6927

closes https://github.com/paradigmxyz/reth/issues/7054

Adds `EtlConfig` which contains file size and **directory**. If a directory is not set, it will do its best to set it to `$DATADIR/etl-tmp`

### CODE REVIEW

File: bin/reth/src/commands/stage/run.rs [hunk 1]
```diff
@@ -155,7 +159,12 @@ impl Command {
 
         let batch_size = self.batch_size.unwrap_or(self.to - self.from + 1);
 
-        let etl_file_size = self.etl_file_size.unwrap_or(500 * 1024 * 1024);
+        let etl_config = EtlConfig::new(
+            Some(
+                self.etl_dir.unwrap_or_else(|| EtlConfig::from_datadir(&data_dir.data_dir_path())),
+            ),
+            self.etl_file_size.unwrap_or(500 * 1024 * 1024),
```
Comment (mattsse) [inline bin/reth/src/commands/stage/run.rs:166]: can we make this a constant?

**Review by mattsse — APPROVED** [review]
smol nit, pending @shekhirin

**Comment by datanexus-vincent** [comment]
Confirmed working on my end as well. Thank you for your time, effort, and the quick turn around!

**Review by shekhirin — APPROVED** [review]
LGTM

**Comment by gakonst** [comment]
@datanexus-vincent thanks for your patience and for highlighting the issue.

**Pull request merged**